src/lightning/fabric/utilities/data.py

```diff
@@ -180,6 +180,7 @@ def _dataloader_init_kwargs_resolve_sampler(
     if batch_sampler is not None and type(batch_sampler) is not BatchSampler:
         batch_sampler_cls = type(batch_sampler)
         if hasattr(batch_sampler, "__pl_saved_args"):
+            # This is a PyTorch `BatchSampler` for which we captured the init args
```

Comment:
This is not a batch sampler, this is a something-else to which we try to inject the PyTorch batch sampler kwargs.

Reply:
No, this must be a batch sampler because we specifically collect the `__pl_saved_args` on subclasses of `BatchSampler`: https://github.com/Lightning-AI/lightning/blob/ed06956c9e6d1a6449ba46d62ee9bd624b5a6a66/src/lightning/pytorch/trainer/connectors/data_connector.py#L337

Reply:
Okay. Then this should clarify my confusion

```suggestion
            # This is a PyTorch `BatchSampler` subclass for which we captured the init args
```